### PR TITLE
feature: dark mode feature added to app

### PR DESCRIPTION
**Pull request by kylecalbert**

### CODE REVIEW

File: src/Components/ThemeContext/ThemeContext.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// ThemeContext.js
+import React, { createContext, useContext, useState } from 'react';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
```
Comment (sukhvir-notta) [inline src/Components/ThemeContext/ThemeContext.js]: This import can be removed

File: src/Components/ThemeContext/ThemeContext.js
Comment (sukhvir-notta) [inline src/Components/ThemeContext/ThemeContext.js:1]: Could `ThemeContext` be moved outside of the components folder?

File: src/Components/header/header.js
Comment (sukhvir-notta) [inline src/Components/header/header.js:1]: Component filename should be capitalised, make sure file and folder naming is consistent

File: src/Components/UserSearch/UserSearch.js [hunk 2]
```diff
@@ -7,8 +7,8 @@ import { Header } from '../Header/Header';
 export const StyledContainer = styled.div`
```
Comment (sukhvir-notta) [inline src/Components/UserSearch/UserSearch.js]: Should the styling be in a separate file similar to `Searchbar`?

File: src/Components/header/header.js [hunk 3]
```diff
@@ -26,17 +36,21 @@ export const IconContainer = styled.div`
...
 
 export const Header = () => {
+  const { theme, toggleTheme } = useTheme();
+
   return (
     <HeaderContainer>
-      <Text size="1.5rem">devfinder</Text>
-      <RightContainer>
-        <Text>Light</Text>
+      <Text color={theme.text} size="1.5rem">
```
Comment (sukhvir-notta) [inline src/Components/header/header.js]: The sizes could be moved into a constants file similar to previous projects if they're being repeated in multiple components

File: src/Components/header/header.js
Comment (kylecalbert) [reply src/Components/header/header.js:1]: not sure why its showing as header.js, on my PC it shows Header.js, could be a bug. But will reoslve

**Pull request merged**